src/NServiceBus.Core/DataBus/DataBusSendBehavior.cs

```diff
@@ -73,7 +72,7 @@ class DataBusSendBehavior : Behavior<OutgoingLogicalMessageContext>
                     }
 
                     //we use the headers to in order to allow the infrastructure (eg. the gateway) to modify the actual key
-                    context.SetHeader("NServiceBus.DataBus." + headerKey,headerValue);
+                    context.Headers["NServiceBus.DataBus." + headerKey] = headerValue;
```

Comment:
I think interpolation just adds more noise in this case?
`context.Headers[$"NServiceBus.DataBus.{headerKey}"] = headerValue`